htmd/builder/builder.py

```diff
@@ -171,33 +182,23 @@ def detectDisulfideBonds(mol, thresh=3):
         A list of :class:`DisulfideBridge <htmd.builder.builder.DisulfideBridge>` objects
     """
     from scipy.spatial.distance import pdist, squareform
-    from scipy.sparse.csgraph import connected_components
-    import pandas as pd
+    from htmd.molecule.molecule import UniqueResidueID
     disubonds = []
 
     # Find all SG atoms belonging to resnames starting with CY
     idx = np.where([(rn[0:2] == 'CY') and (n == 'SG') for rn, n in zip(mol.resname, mol.name)])[0] # 'resname "CY.*" and name SG'
     if len(idx) == 0:
-            return disubonds
+        return disubonds
 
-    # Used only for displaying nice messages
-    if mol.chain is None:
-        chains = [''] * len(idx)
-    else:
-        chains = mol.chain[idx]
-    segids = mol.segid[idx]
-    resids = mol.resid[idx]
-    serials = mol.serial[idx]
-    resnames = mol.resname[idx]
-    if np.any([len(s) == 0 for s in segids]):
+    if np.any([len(s) == 0 for s in mol.segid[idx]]):
         raise RuntimeError('Cannot detect disulfide bonds without segment names defined.')
 
-    # Check that there is only one SG atom in the same resid/segid combo
-    df = pd.DataFrame({'segids': segids, 'resids': resids, 'indexes': idx})
-    groups = df.groupby(['resids','segids']).groups
-    for k in groups:
-        if len(df['indexes'][groups[k]].tolist()) != 1:
-            raise RuntimeError('Multiple SG atoms detected in segment {} resid {}. Can\'t guess disulfide bridges.'.format(k[1], k[0]))
+    residues = [UniqueResidueID.fromMolecule(mol, idx=i) for i in idx]
+    for r1 in range(len(residues)):
+        for r2 in range(r1+1, len(residues)):
+            if residues[r1] == residues[r2]:
+                raise RuntimeError('Multiple SG atoms detected in the same residue {}. '
+                                   'Can\'t guess disulfide bridges.'.format(residues[r1]))
```

Comment:
OK, maybe I'm thinking too much about robustness (I can't help it), but I think this is possible from what I read here: http://www.bmsc.washington.edu/CrystaLinks/man/pdb/part_62.html: there can be two cysteins with the same residue ID but different insertions, and this will make it error here, while it may not be an error at all.

I think we need to bring the `idx` definition above to work with `UniqueAtomID`. What do you think?

Reply:
I don't understand. Two cysteins with insertion code would be different `UniqueResidueID` objects. Or are you talking about `altlocs` (which as I said in the other comment don't exist for us after reading the file)? The comparison here is comparing two `UniqueResidueID` objects, not just their `resid`

Reply:
No, actually here I was just messing what the `residues` variable actually meant (and it's equality). Never mind my nonsense!